test/collect_coverage_mock_test.dart

```diff
@@ -628,5 +628,52 @@ void main() {
         'package:baz/baz.dart': {36, 81},
       });
     });
+
+    test(
+        'Collect coverage, scoped output, library filters, '
+        'handles SourceReports that contain unfiltered ranges', () async {
+      // Regression test for https://github.com/dart-lang/coverage/issues/495
+      final service = _mockService(3, 57);
+      when(service.getSourceReport('isolate', ['Coverage'],
+              forceCompile: true,
+              reportLines: true,
+              libraryFilters: ['package:foo/']))
```

Comment:
Ubernit: consider adding trailing commas for formatting.

Reply:
Done